checker.c

```diff
@@ -1,18 +1,44 @@
 #include <stdio.h>
 #include <assert.h>
 
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
+int temperatureIsOk(float temperature)
+{
+  if(temperature < 0 || temperature > 45) 
+  {
     printf("Temperature out of range!\n");
     return 0;
-  } else if(soc < 20 || soc > 80) {
+  }
+  else
+  {return 1;}
+}
+
+int socIsOk(float soc)
+{
+  if(soc < 20 || soc > 80)
+   {
     printf("State of Charge out of range!\n");
     return 0;
-  } else if(chargeRate > 0.8) {
+  }
+  else
+  {return 1;}
+}
+
+int chargeRateIsOk(float chargeRate)
+{
+  if(chargeRate > 0.8)
+   {
     printf("Charge Rate out of range!\n");
     return 0;
   }
-  return 1;
+  else
+  {return 1;}
+}
+
+int batteryIsOk(float temperature, float soc, float chargeRate) {
+  if (1 < (temperatureIsOk(temperature) + socIsOk(soc) + chargeRateIsOk(chargeRate) ))
```

Comment:
Interesting logic ! kindly explain

Reply:
This logic was meant to check if any failure happend on any of 3 condition and return the value